core/trino-main/src/main/java/io/trino/execution/DropTableTask.java

```diff
@@ -62,23 +62,17 @@ public String getName()
         Session session = stateMachine.getSession();
         QualifiedObjectName originalTableName = createQualifiedObjectName(session, statement, statement.getTableName());
         if (metadata.isMaterializedView(session, originalTableName)) {
-            if (!statement.isExists()) {
-                throw semanticException(
-                        TABLE_NOT_FOUND,
-                        statement,
-                        "Table '%s' does not exist, but a materialized view with that name exists. Did you mean DROP MATERIALIZED VIEW %s?", originalTableName, originalTableName);
-            }
-            return immediateVoidFuture();
+            throw semanticException(
+                    TABLE_NOT_FOUND,
```

Comment:
wrong error code

Reply:
cc @findinpath

Reply:
@findepi: Are you rather leaning here towards `io.trino.spi.StandardErrorCode#UNSUPPORTED_TABLE_TYPE` ?

Reply:
@findepi  there is a `io.trino.spi.StandardErrorCode#GENERIC_USER_ERROR` , do you think that will suffice?

Reply:
I went over the entries from `StandardErrorCode` and found  the error code `NOT_SUPPORTED` to fit best to this user error type of exception. Moreover the message of the exception fits also quite well this exception error code.

Reply:
Disregard please my previous comment.
I was drawn to say that `NOT_SUPPORTED` fits best because of the nature of `DROP` statement can be used for tables/views/MV.
However `NOT_SUPPORTED` is rather used to point out that a specific connector has or has not support for a specific statement.

Reply:
> we cannot say "MV is a table of a wrong type", since it;s not a table at all

We use `TABLE_NOT_FOUND` for tables, views and MVs. We don't have specific `VIEW_NOT_FOUND` and `MV_NOT_FOUND` errors. I don't think these are needed either.

Reply:
> To me GENERIC_USER_ERROR sounds better

What's the difference between this case (an object with different type and same name exists) and case where there is no object with given name?

I cannot rationale why we would like to use `GENERIC_USER_ERROR` here rather than `TABLE_NOT_FOUND`? After all, one can say that trying to drop a table that doesn't exist is also `GENERIC_USER_ERROR`

Reply:
You said that TABLE_NOT_FOUND is used for tables/views/MVs
... so it's not a clean indication that a relation was found, but of a wrong type

Reply:
Ok